tests/suites/hup/suite.js

```diff
@@ -0,0 +1,479 @@
+
+    let attempt = 0;
+    const dlOp = async () => {
+      attempt++;
+      that.log(`Fetching balenaOS version ${version}, attempt ${attempt}...`);
+
+      // TODO take version into account for caching...
+      // how does this work for `latest`?
+      if (await fse.pathExists(path)) {
+        that.log(`[Cached used]`);
+        return path;
+      }
+
+      // TODO progress
+      const stream = await that.context.get().sdk.getDownloadStream(
+          that.suite.deviceType.slug,
+          version,
+        );
+      await new Promise((resolve, reject) => {
+        stream.pipe(fs.createWriteStream(path))
+          .on("finish", resolve)
+          .on("error", reject);
+        });
+
+      return path
+    };
+
+    that.suite.teardown.register(async () => {
+        that.log(`Teardown base image`);
+        try {
+          fse.unlinkSync(path);
+        } catch (err) {
+          that.log(`Error removing base image: ${err}`);
+        }
+      });
+
+    return retry(dlOp, { max_retries: 3, interval: 500 });
+};
+// 2}}}
+
+// configureOS {{{2
+const injectBalenaConfiguration = (image, configuration) => { // {{{3
+  // taken from: https://github.com/balena-io/leviathan/blob/master/core/lib/components/os/balenaos.js#L31
+  return imagefs.interact(image, 1, async (fs) => {
+    return require("util").promisify(fs.writeFile)("/config.json",
+      JSON.stringify(configuration));
+  });
+}; // 3}}}
+const injectNetworkConfiguration = async (image, configuration) => { // {{{3
+  // taken from: https://github.com/balena-io/leviathan/blob/master/core/lib/components/os/balenaos.js#L43
+  if (configuration.wireless == null) {
+    return;
+  }
+  if (configuration.wireless.ssid == null) {
+    throw new Error(
+      `Invalide wireless configuration: ${configuration.wireless}`,
+    );
+  }
+
+  const wifiConfiguration = [
+    '[connection]',
+    'id=balena-wifi',
+    'type=wifi',
+    '[wifi]',
+    'hidden=true',
+    'mode=infrastructure',
+    `ssid=${configuration.wireless.ssid}`,
+    '[ipv4]',
+    'method=auto',
+    '[ipv6]',
+    'addr-gen-mode=stable-privacy',
+    'method=auto',
+  ];
+
+  if (configuration.wireless.psk) {
+    Reflect.apply(wifiConfiguration.push, wifiConfiguration, [
+      '[wifi-security]',
+      'auth-alg=open',
+      'key-mgmt=wpa-psk',
+      `psk=${configuration.wireless.psk}`,
+    ]);
+  }
+
+  await imagefs.interact(image, 1, async (fs) => {
+    return require("util").promisify(fs.writeFile)("/system-connections/balena-wifi",
+      wifiConfiguration.join('\n'));
+  });
+}; // 3}}}
+const configureOS = async (that, imagePath, network, configJson) => {
+  that.log(`Configuring base image`);
+  await injectBalenaConfiguration(imagePath, configJson);
+  await injectNetworkConfiguration(imagePath, network);
+};
+// 2}}}
+
+const runRegistry = async (that, seedWithImage) => { // {{{2
+  const docker = new Docker();
+  const registryImage = 'docker.io/library/registry:2';
```

Comment:
```suggestion
  const registryImage = 'registry:2';
```
The previous path is incredibly flaky for some reason, this consistently worked for me.

Reply:
Resolved it in recent commits